### PR TITLE
Splitscreen bug fixes

### PR DESCRIPTION
**Pull request by RickyMorel**

### CODE REVIEW

File: Assets/Scripts/Combat/Shield.cs [hunk 1]
```diff
@@ -33,13 +33,10 @@ private void OnCollisionStay(Collision collision)
...
 
-        if(LayerMask.LayerToName(collision.gameObject.layer) == "NPC") { CheckForEnemyCollision(collision); return; }
-    }
+        if(LayerMask.LayerToName(collision.gameObject.layer) == "NPC") { CheckForEnemyCollision(collision); }
 
-    private void OnCollisionEnter(Collision collision)
-    {
         //If object layer is one of the crash layers, except for NPC
-        if (_shipHealth.CrashLayers == (_shipHealth.CrashLayers | (1 << collision.gameObject.layer))) { CheckForSceneCollision(collision); return; }
+       // if (_shipHealth.CrashLayers == (_shipHealth.CrashLayers | (1 << collision.gameObject.layer))) { CheckForSceneCollision(collision); }
```
Comment (Arielelnoob115Z) [inline Assets/Scripts/Combat/Shield.cs]: XD

File: Assets/Scripts/Combat/WeaponShoot.cs [hunk 2]
```diff
@@ -12,7 +12,7 @@ public class WeaponShoot : MonoBehaviour
 
     #region Private Variable
 
-    protected Weapon _weapon;
+    [SerializeField] protected Weapon _weapon;
```
Comment (Arielelnoob115Z) [inline Assets/Scripts/Combat/WeaponShoot.cs]: ?

File: Assets/Scripts/Enviroment/SceneLoader.cs [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        LoadScene(_sceneToLoad);
+
+        Ship.Instance.transform.position = _shipPosition;
+
+        _startedLoadingScene = true;
+    }
+
+    public static void LoadScene(int sceneToLoad)
+    {
+        Debug.Log("LoadScene!!!");
```
Comment (Arielelnoob115Z) [inline Assets/Scripts/Enviroment/SceneLoader.cs]: Is this still needed?

File: Assets/Scripts/GameManager/DeathManager.cs [hunk 4]
```diff
@@ -82,12 +82,16 @@ private IEnumerator DeathCoroutine()
 
         yield return new WaitForSeconds(2f);
 
-        SaveSystem.Save(Ship.Instance.ShipData.CurrentSaveIndex);
+        //SaveSystem.Save(Ship.Instance.ShipData.CurrentSaveIndex);
```
Comment (Arielelnoob115Z) [inline Assets/Scripts/GameManager/DeathManager.cs:86]: I assume that you forgot about this.

File: Assets/Scripts/Input/PlayerInputHandler.cs [hunk 5]
```diff
@@ -175,6 +192,8 @@ public void SpecialAction()
 
         if (!_player.GetButtonDown("SpecialAction")) { return; }
 
+        Debug.Log("SpecialAction");
```
Comment (Arielelnoob115Z) [inline Assets/Scripts/Input/PlayerInputHandler.cs]: Are these debug.log needed?

File: Assets/Scripts/Interactable/CraftingStation.cs [hunk 6]
```diff
@@ -76,18 +76,20 @@ private void Craft(CraftingRecipy craftingRecipy, List<ItemQuantity> usedResourc
...
     private void ResetText(string text)
     {
+        Debug.Log($"ResetText: {text}");
+        StopAllCoroutines();
         _craftingText.text = text;
         _shouldCountUp = true;
         _counter = 0;
 
-        if (_lastRoutine != null) { StopCoroutine(_lastRoutine); }
+        //if (_lastRoutine != null) { StopCoroutine(_lastRoutine); StopAllCoroutines(); }
```
Comment (Arielelnoob115Z) [inline Assets/Scripts/Interactable/CraftingStation.cs]: some debugs again and this commented code.

File: Assets/Scripts/Player/PlayerStateMachine.cs [hunk 7]
```diff
@@ -63,9 +64,9 @@ public override void Start()
 
     public void AttachToShip(bool isAttached)
     {
-        _isAttachedToShip = isAttached;
+        //_isAttachedToShip = isAttached;
```
Comment (Arielelnoob115Z) [inline Assets/Scripts/Player/PlayerStateMachine.cs]: You already know

File: Assets/Scripts/Player/PlayerStateMachine.cs [hunk 8]
```diff
@@ -107,15 +108,35 @@ public override void FixedUpdate()
...
+    public void Teleport(Vector3 position)
+    {
+        StartCoroutine(TeleportCoroutine(position));
+    }
+
+    private IEnumerator TeleportCoroutine(Vector3 position)
+    {
+        _characterController.enabled = false;
+
+        yield return new WaitForEndOfFrame();
```
Comment (Arielelnoob115Z) [inline Assets/Scripts/Player/PlayerStateMachine.cs:129]: Can't these be done in the same frame?

File: Assets/Scripts/Ship/ShipFastTravel.cs [hunk 9]
```diff
@@ -16,10 +17,10 @@ public class ShipFastTravel : MonoBehaviour
 
     #region Private Variables
 
-    private int _playersInShip = 0;
+    [SerializeField] private int _playersInShip = 0;
```
Comment (Arielelnoob115Z) [inline Assets/Scripts/Ship/ShipFastTravel.cs]: Serializefield in private variables

File: Assets/Scripts/Ship/ShipFastTravel.cs [hunk 10]
```diff
@@ -149,12 +195,12 @@ public void DetachFromShip()
...
-        for (int i = 0; i < _isPlayerActive.Length; i++)
-        {
-            if (_isPlayerActive[i].IsPlayerActive == true)
-            {
-                _isPlayerActive[i].GetComponentInParent<PlayerStateMachine>().AttachToShip(false);
-            }
-        }
+        //for (int i = 0; i < _isPlayerActive.Length; i++)
+        //{
+        //    if (_isPlayerActive[i].IsPlayerActive == true)
```
Comment (Arielelnoob115Z) [inline Assets/Scripts/Ship/ShipFastTravel.cs]: Damn bro.

File: Assets/Scripts/Ship/ShipFastTravel.cs [hunk 11]
```diff
@@ -83,6 +113,14 @@ private void CheckPlayersInShip()
         AttachToShip();
     }
 
+    private IEnumerator CheckIfCameraNeedsOrthoMode()
```
Comment (Arielelnoob115Z) [inline Assets/Scripts/Ship/ShipFastTravel.cs]: I'm pretty sure when that when fast traveling, it always need to be changed to ortho, some maybe change the name to ChangeToOrthoMode or something.

File: Assets/Scripts/UI/LoadScreenUI.cs [hunk 12]
```diff
@@ -0,0 +1,66 @@
...
+    }
+
+    #endregion
+
+    public void LoadScene(AsyncOperation operation)
+    {
+        StartCoroutine(LoadSceneAsync(operation));
+    }
+
+    private IEnumerator LoadSceneAsync(AsyncOperation operation)
```
Comment (Arielelnoob115Z) [inline Assets/Scripts/UI/LoadScreenUI.cs:50]: Why does this need to be an IEnumarator?

File: Assets/Scripts/Player/PlayerStateMachine.cs (hunk 8 above)
Comment (RickyMorel) [reply Assets/Scripts/Player/PlayerStateMachine.cs:129]: Nope

File: Assets/Scripts/UI/LoadScreenUI.cs (hunk 12 above)
Comment (RickyMorel) [reply Assets/Scripts/UI/LoadScreenUI.cs:50]: So it can be done asynchronously

**Review by Arielelnoob115Z — APPROVED** [review]
This better be the last time.

**Pull request merged**